src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py

```diff
@@ -193,6 +194,65 @@ def set_key(cmd,
         "Fail to set the key '{}' due to a conflicting operation.".format(key))
 
 
+def set_keyvault(cmd,
+                 key,
+                 secret_identifier,
+                 name=None,
+                 label=None,
+                 tags=None,
+                 yes=False,
+                 connection_string=None):
+    connection_string = resolve_connection_string(cmd, name, connection_string)
+    azconfig_client = AzconfigClient(connection_string)
+
+    keyvault_ref_value = json.dumps({"uri": secret_identifier}, ensure_ascii=False)
+    retry_times = 3
+    retry_interval = 1
+
+    label = label if label and label != ModifyKeyValueOptions.empty_label else None
+    for i in range(0, retry_times):
+        try:
+            retrieved_kv = azconfig_client.get_keyvalue(key, QueryKeyValueOptions(label))
+        except HTTPException as exception:
+            raise CLIError(str(exception))
+
+        if retrieved_kv is None:
+            set_kv = KeyValue(key, keyvault_ref_value, label, tags, KEYVAULT_CONTENT_TYPE)
+        else:
+            logger.warning("This operation will result in overwriting existing key whose value is: %s", retrieved_kv.value)
+            set_kv = KeyValue(key=key,
+                              label=label,
+                              value=keyvault_ref_value,
+                              content_type=KEYVAULT_CONTENT_TYPE,
+                              tags=retrieved_kv.tags if tags is None else tags)
+            set_kv.etag = retrieved_kv.etag
+
+        verification_kv = {
+            "key": set_kv.key,
+            "label": set_kv.label,
+            "content_type": set_kv.content_type,
+            "value": set_kv.value,
+            "tags": set_kv.tags
+        }
+        entry = json.dumps(verification_kv, indent=2, sort_keys=True)
+        confirmation_message = "Are you sure you want to set the keyvault reference: \n" + entry + "\n"
```

Comment:
add ensure_ascii=False in json.dumps

Reply:
done

---
In reply to: [358478003](https://github.com/Azure/azure-cli/pull/11571#discussion_r358478003) [](ancestors = 358478003)